File: db4oj/core/src/com/db4o/internal/cs/messages/MObjectSetIndexOf.java
/* Copyright (C) 2006  db4objects Inc.  http://www.db4o.com */

package com.db4o.internal.cs.messages;

import com.db4o.internal.cs.*;
import com.db4o.internal.query.result.*;


/**
 * @exclude
 */
public class MObjectSetIndexOf extends MObjectSet {
	
	public boolean processAtServer(ServerMessageDispatcher serverThread) {
		AbstractQueryResult queryResult = queryResult(serverThread, readInt());
		int id = queryResult.indexOf(readInt()); 
		serverThread.write(Msg.OBJECTSET_INDEXOF.getWriterForInt(transaction(), id));
		return true;
	}

}

File: db4on/core/src/core/com/db4o/internal/cs/messages/MObjectSetIndexOf.cs
namespace com.db4o.@internal.cs.messages
{
	/// <exclude></exclude>
	public class MObjectSetIndexOf : com.db4o.@internal.cs.messages.MObjectSet
	{
		public override bool ProcessAtServer(com.db4o.@internal.cs.ServerMessageDispatcher
			 serverThread)
		{
			com.db4o.@internal.query.result.AbstractQueryResult queryResult = QueryResult(serverThread
				, ReadInt());
			int id = queryResult.IndexOf(ReadInt());
			serverThread.Write(com.db4o.@internal.cs.messages.Msg.OBJECTSET_INDEXOF.GetWriterForInt
				(Transaction(), id));
			return true;
		}
	}
}
